src/fs-gen/src/cli_args.rs

```diff
@@ -0,0 +1,80 @@
+use std::{env, path::PathBuf};
+
+use clap::{command, error::ErrorKind, CommandFactory, Parser};
+use regex::Regex;
+
+use once_cell::sync::Lazy;
+
+// So, for any of you who may be scared, this is the regex from the OCI Distribution Sepcification for the image name + the tag
+static RE_IMAGE_NAME: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*(\/[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*)*:[a-zA-Z0-9_][a-zA-Z0-9._-]{0,127}").unwrap()
```

Comment:
This regex pattern expects an image name in the form <image:tag> and not just <image>. But our function that downloads layers doesn't have a problem with receiving an image name without tag (it will just use 'latest' as tag by default). Maybe we should change the regex to make tag part optional?
`[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*(\/[a-z0-9]+((\.|_|__|-+)[a-z0-9]+)*)*(?::[a-zA-Z0-9_][a-zA-Z0-9._-]{0,127})?`